pkg/storage/ocs_test.go

```diff
@@ -0,0 +1,685 @@
+				ManagedResources: ocsoperatorv1.ManagedResourcesSpec{
+					CephBlockPools: ocsoperatorv1.ManageCephBlockPools{
+						ReconcileStrategy: "manage",
+					},
+					CephFilesystems: ocsoperatorv1.ManageCephFilesystems{
+						ReconcileStrategy: "manage",
+					},
+					CephObjectStoreUsers: ocsoperatorv1.ManageCephObjectStoreUsers{
+						ReconcileStrategy: "manage",
+					},
+					CephObjectStores: ocsoperatorv1.ManageCephObjectStores{
+						ReconcileStrategy: "manage",
+					},
+				},
+				MonDataDirHostPath: "/var/lib/rook",
+				MultiCloudGateway: &ocsoperatorv1.MultiCloudGatewaySpec{
+					ReconcileStrategy: "manage",
+				},
+				StorageDeviceSets: make([]ocsoperatorv1.StorageDeviceSet, 0),
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		namespace           string
+		addToRuntimeObjects bool
+		expectedError       error
+		client              bool
+	}{
+		{
+			name:                "test",
+			namespace:           "openshift-storage",
+			addToRuntimeObjects: true,
+			expectedError:       nil,
+			client:              true,
+		},
+		{
+			name:                "",
+			namespace:           "openshift-storage",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("storageCluster 'name' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "test",
+			namespace:           "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("storageCluster 'namespace' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "ocstest",
+			namespace:           "openshift-storage",
+			addToRuntimeObjects: false,
+			expectedError:       fmt.Errorf("storageCluster object ocstest does not exist in namespace openshift-storage"),
+			client:              true,
+		},
+		{
+			name:                "ocstest",
+			namespace:           "openshift-storage",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("storageCluster 'apiClient' cannot be empty"),
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testStorageCluster := generateStorageCluster(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testStorageCluster)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := PullStorageCluster(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError != nil {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		} else {
+			assert.Equal(t, testStorageCluster.Name, builderResult.Object.Name)
+		}
+	}
+}
+
+func TestNewStorageClusterBuilder(t *testing.T) {
+	generateStorageCluster := NewStorageClusterBuilder
```

Comment:
You can remove this line and use NewStorageClusterBuilder directly in test case.